eva-accession-import/src/test/java/uk/ac/ebi/eva/accession/dbsnp/io/ImportedSubmittedVariantWriterTest.java

```diff
@@ -0,0 +1,159 @@
+
+        List<ImportedSubmittedVariantEntity> accessions = mongoTemplate.find(new Query(),
+                                                                             ImportedSubmittedVariantEntity.class);
+        assertEquals(1, accessions.size());
+        assertEquals(EXPECTED_ACCESSION, (long) accessions.get(0).getAccession());
+
+        assertVariantEquals(variant, accessions.get(0));
+    }
+
+    @Test
+    public void saveDifferentTaxonomies() throws Exception {
+        SubmittedVariant firstSubmittedVariant = new SubmittedVariant("assembly", TAXONOMY_1, "project", "contig",
+                                                                      START_1, "reference", "alternate",
+                                                                      CLUSTERED_VARIANT, false, MATCHES_ASSEMBLY);
+        SubmittedVariant secondSubmittedVariant = new SubmittedVariant("assembly", TAXONOMY_2, "project", "contig",
+                                                                       START_1, "reference", "alternate",
+                                                                       CLUSTERED_VARIANT, false, MATCHES_ASSEMBLY);
+        ImportedSubmittedVariantEntity firstVariant = new ImportedSubmittedVariantEntity(
+                EXPECTED_ACCESSION, hashingFunction.apply(firstSubmittedVariant), firstSubmittedVariant);
+        ImportedSubmittedVariantEntity secondVariant = new ImportedSubmittedVariantEntity(
+                EXPECTED_ACCESSION, hashingFunction.apply(secondSubmittedVariant), secondSubmittedVariant);
+
+        importedSubmittedVariantWriter.write(Arrays.asList(firstVariant, secondVariant));
+
+        List<ImportedSubmittedVariantEntity> accessions = mongoTemplate.find(new Query(),
+                                                                             ImportedSubmittedVariantEntity.class);
+        assertEquals(2, accessions.size());
+        assertEquals(EXPECTED_ACCESSION, (long) accessions.get(0).getAccession());
+
+        assertVariantEquals(firstVariant, accessions.get(0));
+        assertVariantEquals(secondVariant, accessions.get(1));
+    }
+
+    @Test
+    public void failsOnDuplicateVariant() throws Exception {
+        SubmittedVariant submittedVariant = new SubmittedVariant("assembly", TAXONOMY_1, "project", "contig",
+                                                                      START_1, "reference", "alternate",
+                                                                      CLUSTERED_VARIANT, false, MATCHES_ASSEMBLY);
+        ImportedSubmittedVariantEntity variant = new ImportedSubmittedVariantEntity(
+                EXPECTED_ACCESSION, hashingFunction.apply(submittedVariant), submittedVariant);
+
+        thrown.expect(RuntimeException.class);
+        importedSubmittedVariantWriter.write(Arrays.asList(variant, variant));
+    }
+
+    private void assertVariantEquals(ISubmittedVariant expectedvariant, ISubmittedVariant actualVariant) {
```

Comment:
Implement `equals` and `hashCode` in the corresponding classes following [this model](https://github.com/EBIvariation/eva-accession/blob/master/eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/SubmittedVariant.java), then include the extra fields to this method if you want to. But don't compare every single field and leave the classes with no `equals` method!

Reply:
then I'm going to use the SummaryFunction to not replicate the selection of identifying fields

Reply:
Even if you use the summary function here, please still implement `equals` and `hashCode` methods in the corresponding classes. Those do not use the identifying fields like the summary function, but are pure equivalence functions.